pennylane/tape/tapes/tape.py

```diff
@@ -147,7 +147,8 @@ class QuantumTape(AnnotatedQueue):
         caching (int): Number of device executions to store in a cache to speed up subsequent
             executions. A value of ``0`` indicates that no caching will take place. Once filled,
             older elements of the cache are removed and replaced with the most recent device
-            executions to keep the cache up to date.
+            executions to keep the cache up to date. The cache is not available for
+            gradient-based calculations.
```

Comment:
@tom, thanks so much for sorting this out.

I am wondering about this statement though...if a device overwrites `batch_execute` then caching is in principle available. Not sure what is the best formulation here?

Reply:
:thinking: I think even if the device overwrites `batch_execute` then caching is still unavailable - the issue is that the tape's `jacobian()` method now no longer interacts with its own `execute_device()` method (where the caching lives).

Hopefully we can merge in the follow up PR https://github.com/PennyLaneAI/pennylane/pull/851 reasonably soon after merging this one.